tests/tests/test-precompiles.ts

```diff
@@ -22,6 +22,59 @@ describeWithMoonbeam("Moonbeam (Precompiles)", `simple-specs.json`, (context) =>
     "573d6000803e3d6000fd5b5050506040515160601b6bffffffffffffffffffffffff19161461010257600080fd5b" +
     "603f806101106000396000f3fe6080604052600080fdfea26469706673582212202febccafbee65a134279d3397f" +
     "ecfc56a3d2125987802a91add0260c7efa94d264736f6c634300060c0033";
+
+  /*
+   * pragma solidity ^0.7.0;
+   *
+   * contract ModularCheck {
+   *
+   *     // Verify simple modular exponentiation
+   *     constructor() {
+   *         require(modExp(3, 5, 7) == 5);
+   *         require(modExp(5, 7, 11) == 3);
+   *     }
+   *
+   *     // Wrapper function to use the precompile.
+   *     // Taken from https://ethereum.stackexchange.com/a/71590/9963
+   *     function modExp(uint256 _b, uint256 _e, uint256 _m) public returns (uint256 result) {
+   *         assembly {
+   *             // Free memory pointer
+   *             let pointer := mload(0x40)
+   *
+   *             // Define length of base, exponent and modulus. 0x20 == 32 bytes
+   *             mstore(pointer, 0x20)
+   *             mstore(add(pointer, 0x20), 0x20)
+   *             mstore(add(pointer, 0x40), 0x20)
+   *
+   *             // Define variables base, exponent and modulus
+   *             mstore(add(pointer, 0x60), _b)
+   *             mstore(add(pointer, 0x80), _e)
+
+   *             mstore(add(pointer, 0xa0), _m)
+   *             // Store the result
+   *             let value := mload(0xc0)
+   *
+   *             // Call the precompiled contract 0x05 = bigModExp
+   *             if iszero(call(not(0), 0x05, 0, pointer, 0xc0, value, 0x20)) {
+   *                 revert(0, 0)
+   *             }
+   *
+   *             result := mload(value)
+   *         }
+   *     }
+   * }
+   */
+  const MODEXP_CONTRACT_BYTECODE =
```

Comment:
this can be moved to the utils folder

Reply:
I agree. Can we do this in a followup since there are other opportunities to make that improvement (even in this same file) that were not introduced here? I created MOON-234 for it.